bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/BazelRuleAttributes.java

```diff
@@ -23,37 +19,22 @@ public class BazelRuleAttributes {
      * Workaround for https://github.com/bazelbuild/bazel/issues/20918
      */
     private static BinaryOperator<Attribute> firstOneWinsBazelDuplicateWorkaround = (first, second) -> first;
-    private final Rule rule;
-
-    private final Map<String, Attribute> attributesByAttributeName;
+    private final RuleInternal rule;
 
-    BazelRuleAttributes(Rule rule) {
+    BazelRuleAttributes(RuleInternal rule) {
         this.rule = rule;
-        // multiple attributes with the same name are not expected but can happen (https://github.com/bazelbuild/bazel/issues/20918)
-        // we therefore use a merge function which discards all duplicates
```

Comment:
@brian-mcnamara Can you make sure this comment is retained in the new structure somewhere?